crates/core/src/types.rs

```diff
@@ -31,34 +41,364 @@ impl SurfnetTransactionStatus {
+                    .collect(),
+                inner_instructions: Some(
+                    transaction_meta
+                        .inner_instructions
+                        .iter()
+                        .enumerate()
+                        .map(|(i, ixs)| InnerInstructions {
+                            index: i as u8,
+                            instructions: ixs
+                                .iter()
+                                .map(|ix| InnerInstruction {
+                                    instruction: ix.instruction.clone(),
+                                    stack_height: Some(ix.stack_height as u32),
+                                })
+                                .collect(),
+                        })
+                        .collect(),
+                ),
+                log_messages: Some(transaction_meta.logs),
+                pre_token_balances: Some(
+                    pre_token_accounts_with_indexes
+                        .iter()
+                        .zip(token_mints.clone())
+                        .zip(token_program_ids.clone())
+                        .map(|(((i, a), mint), token_program)| TransactionTokenBalance {
+                            account_index: *i as u8,
+                            mint: a.mint.to_string(),
+                            ui_token_amount: UiTokenAmount {
+                                ui_amount: Some(a.amount as f64 / 10f64.powi(mint.decimals as i32)),
+                                decimals: mint.decimals,
+                                amount: a.amount.to_string(),
+                                ui_amount_string: format!(
+                                    "{:.2}",
+                                    a.amount as f64 / 10f64.powi(mint.decimals as i32)
+                                ),
+                            },
+                            owner: a.owner.to_string(),
+                            program_id: token_program.to_string(),
+                        })
+                        .collect(),
+                ),
+                post_token_balances: Some(
+                    post_token_accounts_with_indexes
+                        .iter()
+                        .zip(token_mints)
+                        .zip(token_program_ids)
+                        .map(|(((i, a), mint), token_program)| TransactionTokenBalance {
+                            account_index: *i as u8,
+                            mint: a.mint.to_string(),
+                            ui_token_amount: UiTokenAmount {
+                                ui_amount: Some(a.amount as f64 / 10f64.powi(mint.decimals as i32)),
+                                decimals: mint.decimals,
+                                amount: a.amount.to_string(),
+                                ui_amount_string: format!(
+                                    "{:.2}",
+                                    a.amount as f64 / 10f64.powi(mint.decimals as i32)
```

Comment:
Could we maybe have a little helper for those?

Reply:
Done: 2b9c4af796318fb7c5f9e9793cd13af926551a2c